gameState.cpp

```diff
@@ -2,82 +2,320 @@
   auto rng = std::default_random_engine(time(0));
   std::shuffle(deck.begin(), deck.end(), rng);
-  for (size_t i = 0; i < deck.size(); i++) {
-    std::cout << deck[i] << "\n";
-  }
 
   // Create X numbers of players and distribute cards
   // Push top of deck into each players hands
 
-  for (int i = 0; i < numPlayers; i++) {
+  for (size_t i = 0; i < playerList.size(); i++) {
     std::cout << "Player " << i << " added. \n";
-    card first, second;
+    Card first, second;
     first = deck.back();
     deck.pop_back();
     second = deck.back();
     deck.pop_back();
-    players.at(i).hand = hand(first, second);
-    std::cout << "Player " << i << " : <" << players.at(i).hand.first << ","
-              << players.at(i).hand.second << "> \n";
+    players.at(i)->setHand(Hand(first, second));
   }
 }
 
-void gameState::mainLoop() {
-  // Loop through the player array and get them to do actions
-  // Probability is just randomized.
- while (1) {
-    if (players.size() == 1) {
-      break;
-    }
+Player *gameState::getPlayer(std::string name) {
+  for (auto &player : players)
+    if (player->name == name)
+      return player.get();
+  return nullptr;
+}
+
+void gameState::generalIncome(std::string name) {
+  getPlayer(name)->setCoin(getPlayer(name)->getCoin() + 1);
+}
+
+void gameState::generalAid(std::string name) {
+  //request();
+  getPlayer(name)->setCoin(getPlayer(name)->getCoin() + 2);
+}
+
+// [General Actions]
+void gameState::generalOverthrow(std::string name, std::string target, int pos) {
+  if (getPlayer(target)->isAlive()) {
+    if (getPlayer(target)->hand.first == Card::empty)
+      {
+        deckGraveyard.push_back(getPlayer(target)->hand.second);
+        getPlayer(target)->setAlive(false);
+      }
+    else if (getPlayer(target)->hand.second == Card::empty)
+      {
+        deckGraveyard.push_back(getPlayer(target)->hand.first);
+        getPlayer(target)->setAlive(false);
+      }
+    else
+      {
+        if (pos == 0)
+          {
+            Hand temp = getPlayer(target)->getHand();
+            deckGraveyard.push_back(getPlayer(target)->hand.first);
+            temp.first = Card::empty;
+            getPlayer(target)->setHand(temp);
+          }
+        if (pos == 1)
+          {
+            Hand temp = getPlayer(target)->getHand();
+            deckGraveyard.push_back(getPlayer(target)->hand.second);
+            temp.second = Card::empty;
+            getPlayer(target)->setHand(temp);
+          }
+      }
+  }
+}
+
+// [Baron]
+void gameState::baronTax(std::string name) {
+  // request();
+  getPlayer(name)->setCoin(getPlayer(name)->getCoin() + 3);
+}
+
+void gameState::baronBlockAid(std::string name, std::string target) {
+  // request()
+}
 
-    for (int i = 0; i < players.size(); i++) {
-      if (i == 0) {
-        // Player's choice
+// [Executioner]
+void gameState::executionerExecute(std::string name, std::string target, int pos) {
+  // request();
+  if (getPlayer(target)->isAlive()) {
```

Comment:
The only difference between executionerExecute and generalOverthrow  is `request()`. WDYT about pulling out a function that does the card removal and calling it from both functions.